uncertainty_baselines/models/resnet50_radial.py

```diff
@@ -0,0 +1,281 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+ResNet50 Keras model with Radial variational layers, as introduced in
+Radial Bayesian Neural Networks (Farquhar et al. 2020).
+"""
+
+import functools
+import string
+import warnings
+
+import numpy as np
+import tensorflow as tf
+
+from uncertainty_baselines.models.radial_utils import TrainableHeRadial
+from uncertainty_baselines.models.variational_utils import (
+  get_kernel_regularizer_class, init_kernel_regularizer)
+
+try:
+  import edward2 as ed  # pylint: disable=g-import-not-at-top
+except ImportError as e:
+  warnings.warn(f'Skipped due to ImportError: {e}')
+
+
+BatchNormalization = functools.partial(  # pylint: disable=invalid-name
+  tf.keras.layers.BatchNormalization,
+  epsilon=1e-5,  # using epsilon and momentum defaults from Torch
+  momentum=0.9)
+Conv2DFlipout = functools.partial(  # pylint: disable=invalid-name
+  ed.layers.Conv2DFlipout, use_bias=False)
+
+
+def bottleneck_block(inputs, filters, stage, block, strides, prior_stddev,
+                     dataset_size, stddev_mean_init, stddev_stddev_init,
+                     tied_mean_prior=True):
+  """Residual block with 1x1 -> 3x3 -> 1x1 convs in main path.
+
+  Note that strides appear in the second conv (3x3) rather than the first (1x1).
+  This is also known as "ResNet v1.5" as it differs from He et al. (2015)
+  (http://torch.ch/blog/2016/02/04/resnets.html).
+
+  Args:
+    inputs: tf.Tensor.
+    filters: list of integers, the filters of 3 conv layer at main path
+    stage: integer, current stage label, used for generating layer names
+    block: 'a','b'..., current block label, used for generating layer names
+    strides: Strides for the second conv layer in the block.
+    prior_stddev: Fixed standard deviation for weight prior.
+    dataset_size: Dataset size to properly scale the KL.
+    stddev_mean_init: float, initializes the mean of the TruncatedNormal from
+      which we sample the initial posterior stddev:
+      mean = np.log(np.expm1(stddev_mean_init))
+    stddev_stddev_init: float, stddev of the TruncatedNormal distribution used
+      to initialize the stddev of the variational posterior.
+    tied_mean_prior: bool, if True, fix the mean of the prior to that of the
+      variational posterior, which causes the KL to only penalize the weight
+      posterior's standard deviation, and not its mean.
+
+  Returns:
+    tf.Tensor.
+  """
+  filters1, filters2, filters3 = filters
+  conv_name_base = 'res' + str(stage) + block + '_branch'
+  bn_name_base = 'bn' + str(stage) + block + '_branch'
+  kernel_regularizer_class = get_kernel_regularizer_class(
```

Comment:
I'm not certain, but I think this is calling the standard variational_utils get_kernel_regularizer class which is using whether tied_mean_prior is true to get one of two things that are designed for Normal posterior approximations. Are you sure these are correct here?

Reply:
Yes -- the Ed regularizers end up calling the correct KL between Radial posterior and MVG prior